tiledb/queries/Cargo.toml

```diff
@@ -0,0 +1,12 @@
+[package]
+name = "tiledb-queries"
```

Comment:
I'm not sure I understand why this is a separate crate. Also, the name doesn't strike me as super awesome given how general it is compared to what we've got here.

Reply:
I was envisioning that we might eventually migrate the other query adapters into this crate and leave the API crate with just the barest form of "please fill in my buffer" query API.  Hence the choice of generic name and crate.

Reply:
Maybe I should re-name it to `tiledb-query-adapters`?

Reply:
I'd be +1 for that rename.

Reply:
34ad527